server/vehicle-persistence.lua

```diff
@@ -121,4 +132,50 @@ AddEventHandler('entityRemoved', function(entity)
         local passenger = passengers[i]
         SetPedIntoVehicle(passenger.ped, veh, passenger.seat)
     end
+end)
+
+if persistence == 1 then return end
+
+local function checkVehicleExist(plate)
+    local vehicles = GetGamePool('CVehicle')
+    for i = 1, #vehicles do
+        local vehicle = vehicles[i]
+        if qbx.getVehiclePlate(vehicle) == plate then
+            return true
+        end
+    end
+end
+
+local function spawnVehicle(coords, id, model, props)
+    local _, veh = qbx.spawnVehicle({
+        spawnSource = vector4(coords.x, coords.y, coords.z, coords.w),
+        model = model,
+        props = props
+    })
+    exports.qbx_core:EnablePersistence(veh)
+    Entity(veh).state:set('vehicleid', id, false)
+    SetVehicleDoorsLocked(veh, 2)
+    TriggerClientEvent('qbx_core:client:removeVehZone', -1, id)
+end
+
+lib.callback.register('qbx_core:server:getVehiclesToSpawn', function()
+    local vehicles = {}
+    local query = 'SELECT id, plate, coords FROM player_vehicles WHERE state = 0'
```

Comment:
Rather than being queried by every player. This data should be queried on server startup and cached. Then, as vehicles get spawned, the cache gets smaller and smaller.